service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/PractitionerAgentPersonMapperTest.java

```diff
@@ -0,0 +1,107 @@
+
+import uk.nhs.adaptors.gp2gp.common.service.FhirParseService;
+import uk.nhs.adaptors.gp2gp.common.service.RandomIdGeneratorService;
+import uk.nhs.adaptors.gp2gp.utils.ResourceTestFileUtils;
+import uk.nhs.adaptors.gp2gp.utils.TestArgumentsLoaderUtil;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class PractitionerAgentPersonMapperTest {
+
+    private static final String TEST_ID = "6D340A1B-BC15-4D4E-93CF-BBCB5B74DF73";
+    private static final String PRACTITIONER_FILE_LOCATION = "/ehr/mapper/practitioner/";
+    private static final String PRACTITIONER_ONLY_FILE_LOCATION = PRACTITIONER_FILE_LOCATION + "practitioner-only/";
+    private static final String PRACTITIONER_ROLE_FILE_LOCATION = PRACTITIONER_FILE_LOCATION + "practitioner-role/";
+    private static final String ORGANIZATION = PRACTITIONER_FILE_LOCATION + "organization.json";
+    private static final String PRACTITIONER = PRACTITIONER_FILE_LOCATION + "practitioner.json";
+
+    @Mock
+    private RandomIdGeneratorService randomIdGeneratorService;
+    private PractitionerAgentPersonMapper practitionerAgentPersonMapper;
+    private MessageContext messageContext;
+    private FhirParseService fhirParseService;
+
+    @BeforeEach
+    public void setUp() {
+        when(randomIdGeneratorService.createNewId()).thenReturn(TEST_ID);
+        messageContext = new MessageContext(randomIdGeneratorService);
+        OrganizationToAgentMapper organizationToAgentMapper = new OrganizationToAgentMapper(messageContext);
+        practitionerAgentPersonMapper = new PractitionerAgentPersonMapper(messageContext, organizationToAgentMapper);
+        fhirParseService = new FhirParseService();
+    }
+
+    @ParameterizedTest
+    @MethodSource("readPractitionerRoleTests")
+    public void When_MappingPractitionerToAgent_Expect_RepresentedAgentXml(String practitionerRoleJson, String outputXml)
+        throws IOException {
+
+        var jsonInputPractitionerRole = ResourceTestFileUtils.getFileContent(practitionerRoleJson);
+        var expectedOutput = ResourceTestFileUtils.getFileContent(outputXml);
+
+        Practitioner practitioner = getPractitionerResource();
+        PractitionerRole practitionerRole = fhirParseService.parseResource(jsonInputPractitionerRole, PractitionerRole.class);
+        Organization organization = getOrganizationResource();
+
+        var outputMessage = practitionerAgentPersonMapper
+            .mapPractitionerToAgentPerson(practitioner, Optional.of(practitionerRole), Optional.of(organization));
+        assertThat(outputMessage).isEqualToIgnoringWhitespace(expectedOutput);
+    }
+
+    @ParameterizedTest
+    @MethodSource("readPractitionerOnlyTests")
+    public void When_MappingPractitionerOnlyToAgent_Expect_NoRepresentedOrganization(String inputJson, String outputXml)
+        throws IOException {
+        var jsonInputPractitioner = ResourceTestFileUtils.getFileContent(inputJson);
```

Comment:
If one of the tests fail it is not easy to determine which one is the failing one. Could we extend the test tool to show what test was executed ?  It would be also good if we could execute one single test case without removing files from test directory.